### DISCUSSION TITLE
[Feature Request] Hierarchical note structure

### DISCUSSION THREAD

**Discussion by FranticEr** [SebastianMC/obsidian-custom-sort#70] [Q&A] [answered]
Like OneNote, you can fold child notes under parent notes

**Comment by SebastianMC** [comment, marked as answer]
Hi @FranticEr,

I was a happy user of MS One Note for many many years and I get the idea behind this request.

Let me start with instructions on how you can incorporate the hierarchical note structure without the need of my `custom sort` plugin.
Let me use an ad hoc example to show you the idea.
The steps are:

1. Install the [`alx-folder-note`](https://github.com/aidenlx/alx-folder-note) plugin and its accompanying [`folder-note-core`](https://github.com/aidenlx/folder-note-core) plugin. After installation enable these two plugins and keep their default configuration - specifically the _Note File Storage Strategy_ set to _Inside Folder, With Same Name_
2. In some folder, say a `Hierarchical exercise` folder, create four example notes:
  - `First note`
  - `Second note`
  - `Container note`
  - `Some other note`

You should end up with a structure like below:

![70-s1](https://user-images.githubusercontent.com/23032356/226428680-f606f403-4536-4ae5-801e-6ba7886e9929.jpg)

3. Edit the contents of `Container note` and put some text there, it will help you get the mechanics of the example
4. Now, let's assume that you want the `Container note` to be the parent for grouping other notes. In order to allow structure rearrangement via drag & drop (like in One Note) create the folder `Container note` (Yes, __same name__ as the note, intentionally). You should end up with:
![70-s2](https://user-images.githubusercontent.com/23032356/226429792-e54edb3a-0774-4c63-8852-20f841db8914.jpg)
5. Part one of the magic: drag & drop the note `Container note` into the newly created folder (of same name). The note will seemingly disappear, yet when you activate (click) the `Container note` folder, it will display the content of the note. Nothing is lost, the `Container note` contents is retained. And in result, the OneNote-alike grouping has been enabled for this specific note.
6. The final step, which you know from OneNote: drag & drop the two notes `First note` and `Second note` onto the `Container note`.\
The result should be:
![70-s3](https://user-images.githubusercontent.com/23032356/226430966-76105309-e996-4e5f-ae05-91aaa1180549.jpg)

I'm using the described pattern to work with hierarchical note structure, exactly as I did in OneNote. And it works seamlessly. From conceptual perspective it means that - comparing to OneNote - I only need to explicitly turn a note into a container to enable drag & drop of child notes onto it.

Having the comfort of the above solution, which is close to what OneNote offered, I think (personally) that there is no need to seek any other way for introducing the hierarchy. My `custom-sort` plugin by definition works on the filesystem hierarchy of folders and files and only allows reordering of items in folder(s), not restructuring. At the same time it seamlessly cooperates with the mentioned `alx-folder-note` and `folder-note-core` popular and reliable plugins. That's why I recommend to follow the simple pattern described above
